src/units.cpp

```diff
@@ -278,7 +278,7 @@ ptrdiff_t ulpsDistance(double a, double b)
     }
 
     // If one's infinite and they're not equal, max distance.
-    if (std::isinf(a) xor std::isinf(b)) {
+    if (!std::isinf(a) != !std::isinf(b)) {
```

Comment:
`!X` is used to convert `X` to a `bool`, but `X` is already a `bool` here, so we can get away with just:
```c++
if (std::isinf(a) != std::isinf(b)) { 
```

This being said, the original code does a simple `||` operation not a `xor` one, so why a `xor` operation here?

Reply:
.. because it only matters if they're different?  `||` doesn't give you that?

Reply:
Hmm. indeed, it only matters if they are different. (IOW, the original code could have been improved and that's what @hsorby did. Good good.)

Reply:
The original code wasn't a simple `||`.

Reply:
?? I can't see a `xor` on `std::isinf()` at https://bitbashing.io/comparing-floats.html, or am I missing something? Anyway, I guess it doesn't matter.